modules/desktop-virtualization/host-pools/main.bicep

```
@@ -101,11 +101,56 @@ param startVMOnConnect bool = false
 @sys.description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalIds\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
 param roleAssignments array = []
 
+// begin issue #3523 parameters fix section
+@sys.description('Optional. Enable scheduled agent updates, Default means agent updates will automatically be installed by AVD when they become available.')
+@allowed([
+  'Default'
+  'Scheduled'
+])
+param agentUpdateType string = 'Default'
+
+@sys.description('Optional. Update hour for scheduled agent updates.')
+@minValue(1)
+@maxValue(23)
+param agentUpdateMaintenanceWindowHour int = 22
+
+@sys.description('Optional. Update day for scheduled agent updates.')
+@allowed([
+  'Sunday'
+  'Monday'
+  'Tuesday'
+  'Wednesday'
+  'Thursday'
+  'Friday'
+  'Saturday'
+])
+param agentUpdateMaintenanceWindowDayOfWeek string = 'Sunday'
+
+@sys.description('Optional. List of maintenance windows for scheduled agent updates.')
+param agentUpdateMaintenanceWindows array = [
+  {
+    hour: agentUpdateMaintenanceWindowHour
+    dayOfWeek: agentUpdateMaintenanceWindowDayOfWeek
```

Wouldn't it make sense to have the defaults for this one default maintenance window here? I.e.:
```suggestion
    hour: 22
    dayOfWeek: 'Sunday'
```

As soon as somebody where to introduce a custom (set) of window(s), the other default parameter values woudn't be applied anymore.

The same could be set for the `agentUpdate` object in general. It's default value could be:

```PowerShell
param agentUpdate object = {
  type: agentUpdateType
  maintenanceWindows: [
    {
      hour: 22
      dayOfWeek: 'Sunday'
    }
  ]
  maintenanceWindowTimeZone: 'Central Standard Time'
  useSessionHostLocalTime: false
}
```

---

I've created parameters for those values, so you have the option to change them
```
@sys.description('Optional. Update hour for scheduled agent updates.')
@minValue(1)
@maxValue(23)
param agentUpdateMaintenanceWindowHour int = 22

@sys.description('Optional. Update day for scheduled agent updates.')
@allowed([
  'Sunday'
  'Monday'
  'Tuesday'
  'Wednesday'
  'Thursday'
  'Friday'
  'Saturday'
])
param agentUpdateMaintenanceWindowDayOfWeek string = 'Sunday'
```

---

Come to think of it a bit more, I think you're right. As the `agentUpdate` parameter is an object, this makes sense. My thinking got a bit stuck on that array I commented first.